src/Ui.GraphiQL/Internal/graphiql.cshtml

```diff
@@ -286,7 +275,7 @@
         onEditQuery: onEditQuery,
         onEditVariables: onEditVariables,
         onEditOperationName: onEditOperationName,
-        headerEditorEnabled: @Model.HeaderEditorEnabled,
+        isHeadersEditorEnabled: @Model.HeaderEditorEnabled,
```

Comment:
The `HeaderEditorEnabled` property was ineffective because the react property name was incorrect.  This change fixes the bug.